doctor.py

```diff
@@ -1,55 +1,68 @@
 from __future__ import annotations
 
 import argparse
+from collections.abc import Callable
 from collections.abc import Sequence
 from concurrent.futures import ThreadPoolExecutor
 from pkgutil import walk_packages
 from typing import Dict
+from typing import List
+from typing import Set
+from typing import Tuple
 
-help = "Diagnose common issues, and optionally try to fix them."
+from devenv.lib_check.types import checker
+from devenv.lib_check.types import fixer
 
+help = "Diagnose common issues, and optionally try to fix them."
 
-def main(context: Dict[str, str], argv: Sequence[str] | None = None) -> int:
-    parser = argparse.ArgumentParser(description=help)
-    parser.add_argument(
-        "--tag", type=str, action="append", help="Used to match a subset of checks."
-    )
-    parser.add_argument("--check-only", action="store_true", help="Do not run fixers.")
-    args = parser.parse_args(argv)
 
-    match_tags = set(args.tag if args.tag else ())
+class Check:
+    def __init__(
+        self,
+        name: str,
+        tags: Set[str],
+        check: Callable[[], Tuple[bool, str]],
+        fix: Callable[[], Tuple[bool, str]],
+    ):
+        self.name = name
+        self.tags = tags
+        self.check = checker(check)
+        self.fix = fixer(fix)
 
-    repo = context["repo"]
-    if repo not in {"sentry", "getsentry", "devenv"}:
-        print(f"repo {repo} not supported yet!")
-        return 1
 
+def load_checks(context: Dict[str, str], match_tags: Set[str]) -> List[Check]:
     checks = []
     for module_finder, name, ispkg in walk_packages((f'{context["reporoot"]}/devenv/checks',)):
         module = module_finder.find_spec(name).loader.load_module(name)  # type: ignore
         assert isinstance(module.name, str)
         assert isinstance(module.tags, set)
+        assert hasattr(module, "check")
```

Comment:
Optional: This is just an idea, but moving the assertions into the Check __init__ function would simplify the load_checks logic and I think be a little easier to follow the code

For example:

```
checks.append(Check(module))



class Check:
    def __init__(module):
        assert isinstance(module.name, str)
        name = module.name
        
        assert isinstance(module.tags, set)
        tags = module.tags
        
        ........
```

Reply:
One problem with that would be the duplicate logic needed to check the tags inside of load_checks since we use the tags to decide whether a check should be filtered out.

Reply:
Yeah sorry I guess I'd go with:

```
c = Check(module)
if match_tags && c.tags.issubset(match_tags):
	checks.append(c)
```

I was largely thinking that Check is responsible to validating input to create a check rather than the loader, but it's questionable if it's any better.

Reply:
I see, that works!